### PR TITLE
[IMP] locale: add start week

### PR DESCRIPTION
**Pull request by pro-odoo**
This commit adds the start week to the locale. It will be useful to introduce the granularity `day_of_week` for the pivots.

Task: 4031965

## Description:

description of this task, what is implemented and why it is implemented that way.

Task: : [TASK_ID](https://www.odoo.com/web#id=TASK_ID&action=333&active_id=2328&model=project.task&view_type=form&cids=1&menu_id=4720)

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_t("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

### CODE REVIEW

File: tests/model/model_import_export.test.ts [hunk 1]
```diff
@@ -766,6 +772,21 @@ test("import localized date as string and detect the format", () => {
   expect(getEvaluatedCell(model, "A1")?.formattedValue).toBe("31/12/2020");
 });
 
+test("Week start is automatically added during migration", () => {
+  expect(
+    new Model({
+      version: 17,
```
Comment (rrahir) [inline tests/model/model_import_export.test.ts:778]: some reason to target 17 and not 19 ?

File: src/migrations/locale.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
+export const WEEK_START = {
```
Comment (rrahir) [inline src/migrations/locale.ts:1]: Could add a FIXME to remove this map once FF starts supporting https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/getWeekInfo

**Review by rrahir — COMMENTED** [review]
WDYT of taking the same strategy as the currencies and have a dedicated object that would be implemented by the  integrator instead of having to support everything in the library directly?

**Comment by pro-odoo** [comment]
> WDYT of taking the same strategy as the currencies and have a dedicated object that would be implemented by the integrator instead of having to support everything in the library directly?

Not sure to understand what you mean...

File: tests/model/model_import_export.test.ts (hunk 1 above)
Comment (pro-odoo) [reply tests/model/model_import_export.test.ts]: When I started this task, we were on version 17 :D

File: doc/integrating/integration.md [hunk 3]
```diff
@@ -139,6 +139,7 @@ const locale = {
     code: "en_US",
     thousandsSeparator: ",",
     decimalSeparator: ".",
+    weekStart: 7,
```
Comment (rrahir) [inline doc/integrating/integration.md:142]: the 1-7 mapping could be explained in the doc as well

Comment (rrahir) [reply doc/integrating/integration.md:142]: Also maybe add that information in the Localisation sidepanel?

Comment (pro-odoo) [reply doc/integrating/integration.md]: Updated :)

**Comment by rrahir** [comment]
@robodoo r+

**Pull request closed without merging**